course-mw/enrolment-actor/src/test/scala/org/sunbird/enrolments/CourseConsumptionActorTest.scala

```diff
@@ -22,7 +23,94 @@ import scala.concurrent.duration.FiniteDuration
 class CourseConsumptionActorTest extends FlatSpec with Matchers with MockFactory {
     implicit val ec: ExecutionContext = ExecutionContext.global
     val system = ActorSystem.create("system")
-    
+
+    "get Consumption with progressDetails" should "return response" in {
+        val cassandraOperation = mock[CassandraOperation]
+
+        val progressResponse = new java.util.HashMap[String, AnyRef]()
+        progressResponse.put("key1", "val1")
+        progressResponse.put("key2", "val2")
+
+        val response = new Response()
+        response.put("response", new java.util.ArrayList[java.util.Map[String, AnyRef]] {{
+            add(new java.util.HashMap[String, AnyRef] {{
+                put("userId", "user1")
+                put("courseId", "do_123")
+                put("batchId", "0123")
+                put("contentId", "do_456")
+                put("progressdetails", "{}")
+                put("progressDetails", progressResponse)
+            }})
+        }})
+
+        ((requestContext: RequestContext, keyspace: _root_.scala.Predef.String, table: _root_.scala.Predef.String, filters: _root_.java.util.Map[_root_.scala.Predef.String, AnyRef], fields: _root_.java.util.List[_root_.scala.Predef.String]) => cassandraOperation.getRecords(requestContext, keyspace, table, filters, fields)).expects(*,*,*,*,*).returns(response)
+        val result = callActor(getStateReadRequestWithProgressField(), Props(new ContentConsumptionActor().setCassandraOperation(cassandraOperation, false)))
+        assert(null!= result)
```

Comment:
please add more assertion on the result object to check the progressDetails attribute value.

Reply:
Resolved